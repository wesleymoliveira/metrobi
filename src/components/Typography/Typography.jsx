import * as S from './Typography.styled';

export function Typography({
	children,
	color = 'white',
	size = 'medium',
	as = 'h1'
}) {
	return (
		<S.Wrapper
			children
			data-testid='typography'
			color={color}
			size={size}
			as={as}
		>
			{children}
		</S.Wrapper>
	);
}
