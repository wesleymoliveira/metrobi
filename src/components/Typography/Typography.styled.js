import styled, { css } from "styled-components";

export const wrapperModifiers = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.small};
  `,

  medium: (theme) => css`
    font-size: ${theme.font.sizes.medium};
  `,

  large: (theme) => css`
    font-size: ${theme.font.sizes.large};
  `,

  xlarge: (theme) => css`
    font-size: ${theme.font.sizes.xlarge};
  `,

  xxlarge: (theme) => css`
    font-size: ${theme.font.sizes.xxlarge};
  `,
};

export const Wrapper = styled.h1`
  ${({ theme, color, size }) => css`
    color: ${theme.colors[color]};
    ${!!size && wrapperModifiers[size](theme)};
  `}
`;
