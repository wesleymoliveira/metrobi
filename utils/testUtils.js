import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

import theme from "../src/styles/theme";

const AllTheProviders = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui, options) => {
  return render(ui, { wrapper: AllTheProviders, ...options });
};

const snapshotRenderer = (component) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
};

export * from "@testing-library/react";

export { customRender as render, snapshotRenderer  };
