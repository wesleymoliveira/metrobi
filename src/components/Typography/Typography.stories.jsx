import React from 'react';
import { Typography } from './Typography';

const TypographyArgTypes = {
  label: {
    name: 'label',
    type: { name: 'string', required: false },
    description: 'overwritten description',
    table: {
      type: { 
        summary: 'something short', 
        detail: 'something really really long' 
      },

    },
    control: {
      type: 'string'
    }
  }
};

export default {
	title: 'components/Typography',
  parameters: {
    backgrounds: {
      default: 'dark',
    },
  },
	component: Typography,
	argTypes: TypographyArgTypes
};

export const Default = (args) => 
  <div style={{ display: 'flex', flexDirection: 'column'}}>
    <Typography {...args}>Default</Typography>
    <Typography {...args} size='small'>Small</Typography>
    <Typography {...args} size='medium'>Medium</Typography>
    <Typography {...args} size='large'>Large</Typography>
    <Typography {...args} size='xlarge'>XLarge</Typography>
    <Typography {...args} size='xxlarge'>XXLarge</Typography>
  </div>
