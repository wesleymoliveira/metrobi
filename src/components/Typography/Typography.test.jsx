import { screen } from '@testing-library/react';

import { Typography } from './Typography';
import { render, snapshotRenderer } from '../../../utils/testUtils';


describe('<Typography />', () => {
		it('should render the component', () => {
		render(<Typography />);
		
		expect(screen.getByTestId(/typography/i)).toBeInTheDocument();
	});

	it('should match snapshot', () => {
		const tree = snapshotRenderer(<Typography />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('should match the original sizes', () => {
		const standard = snapshotRenderer(<Typography>Default</Typography>).toJSON();

		const small = snapshotRenderer(<Typography size='small'>Small</Typography>).toJSON();

		const large = snapshotRenderer(<Typography size='large'>Large</Typography>).toJSON();

		const xLarge = snapshotRenderer(<Typography size='xlarge'>xLarge</Typography>).toJSON();

		const xxLarge = snapshotRenderer(<Typography size='xxlarge'>xxLarge</Typography>).toJSON();


		
		expect(standard).toHaveStyleRule('font-size', '1.6rem');
		expect(small).toHaveStyleRule('font-size', '1.4rem');
		expect(large).toHaveStyleRule('font-size', '1.8rem');
		expect(xLarge).toHaveStyleRule('font-size', '2.0rem');
		expect(xxLarge).toHaveStyleRule('font-size', '2.8rem');
	});

});
