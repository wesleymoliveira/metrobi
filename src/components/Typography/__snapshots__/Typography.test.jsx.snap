// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> should match snapshot 1`] = `
.c0 {
  color: #FAFAFA;
  font-size: 1.6rem;
}

<h1
  className="c0"
  color="white"
  data-testid="typography"
  size="medium"
/>
`;
